### PR TITLE
Use auth info (if present) when updating pipelines

### PR DESCRIPTION
**Pull request by bidhan-a**
I was getting a lot of these errors since my pipelines are in a git repo which requires authetication

screenshot

This PR should fix that.

Please let me know your thoughts.

Thanks!

### CODE REVIEW

**Comment by Skarlso** [comment]
Looks good! Can you add tests please? :) The policy is, that no code change or code goes in without tests. :)

**Comment by Skarlso** [comment]
@bidhan-a hmm, you'll have to take a look at the git repo framework. see how they do it in their test cases.

Ultimately you can do an integration test with actual credentials. Create a test user and test repo for it. But that would a last resort I think.

Otherwise, this might be a good chance to do #44 :D

**Comment by bidhan-a** [comment]
I couldn't find any test case in go-git which handles repos that require authentication :confused:

It seems like we'll have to look into other options.

**Comment by Skarlso** [comment]
@bidhan-a Oh wait. I think we actually don't have to. Just test this function in the three ways:
https://github.com/gaia-pipeline/gaia/pull/62/files#diff-8b0b0b31212c3aeba36c22017d34614dR144

First, if you provide a username and password the authentication will be that. If not, than there is a public key auth and if neither then nil, nil is returned. We don't have to test github authentication process in this PR because frankly, we don't care. It's a third party component.

We can think about an integration test, but that's minuscule compared to three things we can test now. How does that sound?

**Comment by bidhan-a** [comment]
@Skarlso Yep, that sounds good to me. Thanks! :grin:

File: pipeline/git_test.go [hunk 1]
```diff
@@ -156,3 +156,60 @@ func TestUpdateAllPipelinesHundredPipelines(t *testing.T) {
 		t.Fatal("log output did not contain error message that the repo is up-to-date.: ", b.String())
 	}
 }
+
+func TestGetAuthInfo(t *testing.T) {
```
Comment (Skarlso) [inline pipeline/git_test.go]: Would you be so kind as to extract these into three smaller tests? Or, if you want the setup, than use `t.Run("bla", func() {})` in order to have three legs in one test while still using the setups.

**Comment by michelvocks** [comment]
Good job, @bidhan-a  🤗 In my opinion this is ready to merge. @bidhan-a @Skarlso any objections?

**Pull request merged**